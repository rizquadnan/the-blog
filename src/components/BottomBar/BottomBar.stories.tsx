import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { Bottombar } from './BottomBar'
import { Box } from '@chakra-ui/react'
import { action } from '@storybook/addon-actions'

type TBottombar = React.ComponentProps<typeof Bottombar>
export default {
  title: 'Layout/Bottombar',
  component: Bottombar,
} as Meta<TBottombar>

const Template: Story<TBottombar> = (args) => (
  <Box position="fixed" bottom={0} left={0} right={0}>
    <Bottombar {...args} />
  </Box>
)

export const Default = Template.bind({})
Default.args = {
  selectedNav: 'post',
  onLogout: action('logout'),
}
