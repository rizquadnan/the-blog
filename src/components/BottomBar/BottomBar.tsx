import { Flex, Text } from '@chakra-ui/layout'
import { ReactNode } from 'react'
import {
  HomeIcon,
  PowerIcon,
  SettingsIcon,
  UserGroupIcon,
} from '@/components/uiKit/Icons'
import { VStack } from '@/components/uiKit/VStack'
import { Link } from '@/components/uiKit/Link'
import { useColorModeValue } from '@chakra-ui/react'
import { ColorModeToggle } from '@/components/ColorModeToggle'

type TBottombarItem = {
  icon: ReactNode
  label: string
  isSelected?: boolean
  onClick?(): void
}

function BottombarItem(props: TBottombarItem) {
  const color = useColorModeValue('black', 'white')

  return (
    <VStack
      as="button"
      alignItems="center"
      color={props.isSelected ? 'orange.600' : color}
      transition="color 250ms"
      cursor="pointer"
      _hover={{
        color: 'orange.600',
      }}
      onClick={props.onClick}
    >
      {props.icon}
      <Text fontSize="10px">{props.label}</Text>
    </VStack>
  )
}

type TNavItems = 'post' | 'users' | 'account' | 'logout'
type TBottombar = {
  selectedNav: TNavItems
  onLogout: () => void
}

export function Bottombar(props: TBottombar) {
  const boxShadowColor = useColorModeValue(
    'rgba(149, 157, 165, 0.2)',
    'rgba(149, 157, 165, 0.05)',
  )

  const background = useColorModeValue('white', 'black')

  return (
    <Flex
      w="100%"
      justifyContent="space-between"
      overflowX="scroll"
      padding="16px 20px"
      boxShadow={`${boxShadowColor} 6px 6px 60px`}
      background={background}
    >
      <ColorModeToggle />
      <Link href="/posts" style={{ textDecoration: 'none' }}>
        <BottombarItem
          icon={<HomeIcon fontSize="20px" />}
          label="Posts"
          isSelected={props.selectedNav === 'post'}
        />
      </Link>
      <Link href="/users" style={{ textDecoration: 'none' }}>
        <BottombarItem
          icon={<UserGroupIcon fontSize="20px" />}
          label="Users"
          isSelected={props.selectedNav === 'users'}
        />
      </Link>
      <Link href="/my-account" style={{ textDecoration: 'none' }}>
        <BottombarItem
          icon={<SettingsIcon fontSize="20px" />}
          label="My Account"
          isSelected={props.selectedNav === 'account'}
        />
      </Link>
      <BottombarItem
        onClick={props.onLogout}
        icon={<PowerIcon fontSize="20px" />}
        label="Logout"
        isSelected={props.selectedNav === 'logout'}
      />
    </Flex>
  )
}

export default Bottombar
