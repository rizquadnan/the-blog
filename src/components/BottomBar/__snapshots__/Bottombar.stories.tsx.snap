// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout/Bottombar Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Dark Mode",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": Array [
        Object {
          "checked": undefined,
          "children": undefined,
          "name": "Home",
          "pressed": undefined,
          "role": "button",
          "value": undefined,
        },
      ],
      "name": "Home",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": Array [
        Object {
          "checked": undefined,
          "children": undefined,
          "name": "Users",
          "pressed": undefined,
          "role": "button",
          "value": undefined,
        },
      ],
      "name": "Users",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": Array [
        Object {
          "checked": undefined,
          "children": undefined,
          "name": "My Account",
          "pressed": undefined,
          "role": "button",
          "value": undefined,
        },
      ],
      "name": "My Account",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Logout",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "layout-bottombar--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
